src/components/ListingDetails/index.tsx

```diff
@@ -0,0 +1,323 @@
+          {
+            label: 'Client Location',
+            getValue: (data) =>
+              data.client_location || 'N/A',
+          },
+        ];
+      }
+    } else if (listingData.listing_type === 'INT') {
+      fields = intepretationFields;
+    } else if (listingData.listing_type === 'DOC') {
+      fields = docFields;
+    } else {
+      fields = lcaFields;
+    }
+
+    return (
+      <InnerFieldContainer>
+        {fields.map(({ label, getValue }) => (
+          <div key={label}>{renderField(label, getValue(listingData))}</div>
+        ))}
+      </InnerFieldContainer>
+    );
+  }, [listingData, interpretation]);
+
+  const dateComponent = () => {
+    if (listingData.listing_type !== 'INT') {
+      let dateHeader = 'Assignment Deadline';
+      let dateData = '';
+      if (listingData.listing_type === 'CASE') {
+        dateHeader = 'Next Court/Filing Date';
+        dateData = formatTimestamp(listingData.upcoming_date); // listingData.upcoming_date ? formatTimestamp(listingData.upcoming_date) : "N/A"
+      } else {
+        dateData = formatTimestamp(listingData.deadline);
+      }
+      return (
+        <IconTextGroup>
+          <Icon type="calendar" />
+          <DateText $bold>
+            {dateHeader}: <DateText> {dateData}</DateText>
+          </DateText>
+        </IconTextGroup>
+      );
+    }
+    return null;
+  };
+
+  const profile = useContext(ProfileContext);
+
+  const Interest = useMemo(() => {
+    if (profile && profile.userId)
+      return profile.profileData ? (
+        <InterestForm caseData={listingData as CaseListing} />
+      ) : (
+        <>
+          <H3>
+            Please finish submitting your profile before submitting interest.
+          </H3>
+          <AuthButtons>
+            <LinkButton
+              $primaryColor={COLORS.blueMid}
+              $secondaryColor={COLORS.blueDark}
+              href="/onboarding/roles"
+            >
+              Go to Onboarding
+            </LinkButton>
+          </AuthButtons>
+        </>
+      );
+
+    return (
+      <>
+        <H3>
+          You aren&apos;t signed in. In order to submit your interest, log into
+          your account, or create a new account.
+        </H3>
+        <AuthButtons>
+          <LinkButton $secondaryColor={COLORS.blueMid} href="/signup">
+            Sign Up
+          </LinkButton>
+
+          <LinkButton
+            $primaryColor={COLORS.blueMid}
+            $secondaryColor={COLORS.blueDark}
+            href="/login"
+          >
+            Log In
+          </LinkButton>
+        </AuthButtons>
+      </>
+    );
+  }, [listingData, profile]);
+
+  return (
+    <CaseDisplay>
+      <InfoContainer>
+        <BorderedSection>
+          {dateComponent()}
+          <Flex $align="center" $gap="18px" $direction="row">
+            <img src={logo.src} alt="DOHS Logo" width="77" height="77" />
+            <H2>{listingData.title || 'Migrant seeking representation'}</H2>
```

Comment:
small issue: I wasn't sure if we wanted to keep this default title for assignments. My understanding is that all the case titles default to "Migrant Seeking representation" if they are untitled on LegalServer. However, the title field for `cases` table is nullable on Supabase, so I'm wondering if there's ever a case where a title field would be entry; in that case, I will leave this here.

Reply:
Since admin upload LCA, Doc Translation and One-Time Interpretation requests, all these listings should technically not need a default title since we make it a required field on Retool upon submission